src/com/palmergames/util/Trie.java

```diff
@@ -45,85 +43,75 @@ public Trie() {
 	 * Adds and links new TrieNodes to the trie for each character in the string
 	 * 
 	 * @param key key to add to trie, can be longer than one character
-	 * @return true if the key was added properly, otherwise false 
 	 */
-	public boolean addKey(String key) {
+	public void addKey(String key) {
 		// Current trieNode to crawl through
 		TrieNode trieNode = root;
 
 		// Loop through each character of key
+		outer:
 		for (int i = 0; i < key.length(); i++) {
 			char index = key.charAt(i);
 
 			TrieNode lastNode = trieNode;
-			Optional<TrieNode> optional = lastNode.children.stream()
-				        .filter(e -> e.character == index).findFirst();
-
-			if (!optional.isPresent()) { // No existing TrieNode here, so make a new one
-				trieNode = new TrieNode(index);
-				lastNode.children.add(trieNode); // Put this node as one of lastNode's children
-			} else {
-				trieNode = optional.get();
+			for (TrieNode node : lastNode.children) {
+				if (node.character == index) {
+					trieNode = node;
+					continue outer;
```

Comment:
I am against using labels where preventable because of readability. I believe this label can be avoided by setting `trieNode` to null before the for loop and then if it is null after the loop then you can create the new TrieNode and add it to the last node.

Reply:
Changed, performance remains unaffected